src/framed_write.rs

```diff
@@ -101,7 +141,18 @@ where
 {
     type Error = T::Error;
 
-    fn poll_ready(self: Pin<&mut Self>, _cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+    fn poll_ready(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        let this = &mut *self;
+        while this.buffer.len() >= this.high_water_mark {
+            let num_write = ready!(Pin::new(&mut this.inner).poll_write(cx, &this.buffer))?;
+
+            if num_write == 0 {
+                return Poll::Ready(Err(err_eof().into()));
+            }
+
+            let _ = this.buffer.split_to(num_write);
```

Comment:
Wouldn't it be better to use [`BytesMut::advance`](https://docs.rs/bytes/0.4.12/bytes/struct.BytesMut.html#method.advance) here?

Reply:
Yes. Done in 7f05fc8c1c08ce776dd3ea563511cbc12d47fe30.